internal/provider/lxd/storage.go

```diff
@@ -559,7 +559,7 @@ func (s *lxdFilesystemSource) ImportFilesystem(
 	}
 	volume, eTag, err := s.env.server().GetStoragePoolVolume(lxdPool, storagePoolVolumeType, volumeName)
 	if err != nil {
-		common.HandleCredentialError(callCtx, s.env.credentialInvalidator, IsAuthorisationFailure, err)
+		_, _ = common.HandleCredentialError(callCtx, s.env.credentialInvalidator, IsAuthorisationFailure, err)
```

Comment:
You know, I would rather implement something that is _embedded_ into the providers, rather than calling back out to some _common_ package, passing environs or their components.

This goes for all the other _provider/common_ functionality.

Reply:
This is already exploding more than I want... I can do that, it just makes this task bigger.

I'll do what you want but I'll land this one and then do a follow up PR.

Reply:
It's going to be easier in the long run if I just do it now. So I did.

https://github.com/juju/juju/pull/19007/commits/13cf22af882368b39b9f20ac675c7ac324d9236c#diff-06c7dabd8b58b482a7cd481a08400cb4998cf79aafb3798a374b87b1fbba1ca2R29-R64

The latest commit uses the embedded credential invalidator.

Reply:
I'm not 100% sure it should be in providers/common, but we can always move it. I'm not sold on it being in environs either, but I can be persuaded otherwise.

Reply:
Yeah, I've just realized it should be in environs ( :facepalm: ), but I can't move it there until we've done all the providers and we can delete the HandleCredentialError function.